### PR TITLE
Saving the images and callback

### PR DESCRIPTION
**Pull request by FrederickChan**
Do it after your save.

### CODE REVIEW

**Comment by Catzenjaeger** [comment]
hmn this dosent work. I have try it with two submits. The first submit was correct (all images have id 1) but the secound are wrong. the secound oimage have a wrong id (id 3)

**Comment by Catzenjaeger** [comment]
and new test .. here are the same error. wrong ids

**Comment by FrederickChan** [comment]
Ah yes, i forgot about the loop save. Please change..

File: infusions/figurelib/submit.php [hunk 1]
```diff
@@ -111,15 +111,20 @@
 				"figure_accessories"  => addslash(nl2br(parseubb(stripinput($_POST['figure_accessories'])))),			
 			);
 			
-		// Image upload
-        if (!empty($_FILES['figure_image']) && defender::safe()) {
+		// Insert Figure data 
+		if (defender::safe()) {
+				dbquery_insert(DB_FIGURE_ITEMS, $submitdata, "save", array("keep_session"=>true));
+
```
Comment (FrederickChan) [inline infusions/figurelib/submit.php]: $unique_id = dblastid();

File: infusions/figurelib/submit.php [hunk 2]
```diff
@@ -111,15 +111,20 @@
...
+				
+				// Image upload
+        if (!empty($_FILES['figure_image'])) {
             // To pair with the image into the database.
             $image_data = (array) form_sanitizer($_FILES['figure_image'], "", "figure_image");
             if (!empty($image_data)) {
                 foreach($image_data as $image) {
                     $i_data = array(
                         "figure_images_image_id" => 0,
-                        "figure_images_figure_id" => dblastid(),
```
Comment (FrederickChan) [inline infusions/figurelib/submit.php]: "figure_images_figure_id" => $unique_id,

**Comment by Catzenjaeger** [comment]
thanx its look better .. i make some tests but atm all id's are correct :)

small notice:

figurelib/submit.php
/test/infusions/figurelib/submit.php?stype=f
Undefined variable: unique_id
Line: 123


`		// Insert Figure data 
		if (defender::safe()) {
				dbquery_insert(DB_FIGURE_ITEMS, $submitdata, "save", array("keep_session"=>true));
				
		$unique_id = dblastid();		
				// Image upload
        if (!empty($_FILES['figure_image'])) {
            // To pair with the image into the database.
            $image_data = (array) form_sanitizer($_FILES['figure_image'], "", "figure_image");
            if (!empty($image_data)) {
                foreach($image_data as $image) {
                    $i_data = array(
                        "figure_images_image_id" => 0,
                       // "figure_images_figure_id" => dblastid(), // This will be the primary_key of your last saved record
                       "figure_images_figure_id" => $unique_id,
					   "figure_images_image" => $image['image_name'],
                        "figure_images_thumb" => $image['thumb1_name'],
                        "figure_images_sorting" => "",
                        "figure_images_language" => LANGUAGE,
                    );
                    // Insert image data
					  dbquery_insert(DB_FIGURE_IMAGES, $i_data, "save", array("keep_session"=>true));
                }
            }
        }`

**Comment by FrederickChan** [comment]
figurelib/submit.php
/test/infusions/figurelib/submit.php?stype=f
Undefined variable: unique_id
Line: 123

You had that notice while refreshing before completing the working codes. I believe it's irrelevant to our problems.

**Pull request closed without merging**